src/java.base/unix/classes/sun/nio/fs/UnixFileSystemProvider.java

```diff
@@ -431,10 +431,20 @@ public void createDirectory(Path obj, FileAttribute<?>... attrs)
             dfd2 = dup(dfd1);
             dp = fdopendir(dfd1);
         } catch (UnixException x) {
-            if (dfd1 != -1)
-                UnixNativeDispatcher.close(dfd1);
-            if (dfd2 != -1)
-                UnixNativeDispatcher.close(dfd2);
+            if (dfd1 != -1) {
+                try {
+                    UnixNativeDispatcher.close(dfd1);
+                } catch (UnixException y) {
+                    x.addSuppressed(y);
```

Comment:
Hello Brian,
I think, even after adding this as suppressed exception, this (suprressed) exception will be "lost" because at the end of this outer catch block there's this code:
```
x.rethrowAsIOException(dir);
```
and the implementation of `rethrowAsIOException` internally just creates a new exception instance (depending on the error code) but doesn't capture any suppressed exceptions from the original `x`.

Reply:
Good point.